README.md

```diff
@@ -1,5 +1,98 @@
-# sda
+# Sensitive Data Archive
 
 `SDA` contains all components of [NeIC Sensitive Data Archive](https://neic-sda.readthedocs.io/en/latest/) It can be used as part of a [Federated EGA](https://ega-archive.org/federated) or as a isolated Sensitive Data Archive.
 
 For more information about the different components see the readme files in the respecive folders.
+
+## Developing components of the SDA stack
+
+If you wish to work on the SDA stack itself you'll first need [Go](https://www.golang.org/) installed on your machine. The recommended version can be checked by running:
+
+```sh
+$ make go-version-check
+...
+```
+
+For local dev first make sure Go is properly installed, including setting up a [GOPATH](https://golang.org/doc/code.html#GOPATH). Ensure that $GOPATH/bin is in your path as some distributions bundle the old version of build tools. Next, clone this repository. SDA uses [Go Modules](https://github.com/golang/go/wiki/Modules), so it is recommended that you clone the repository outside of the GOPATH. You can then download any required build tools by bootstrapping your environment:
+
+```sh
+$ make bootstrap
+...
+```
+
+### Makefile options
+
+The Makefile is primarily designd to be an aid during development work.
+
+#### Building the containers
+
+To build all containers for the SDA stack:
+
+```sh
+$ make build-all
+...
+```
+
+To build the container for a speciffic component replace `all` with the folder name:
+
+```sh
+$ make build-<folder-name>
+...
+```
+
+#### Running the integration tests
+
+This will build the container and run the integration test for the postgresql container. The same test will run on every PR in github:
```

Comment:
```suggestion
This will build the container and run the integration test for the PostgreSQL container. The same test will run on every PR in github:
```